pymeasure/instruments/signalrecovery/dsp7265.py

```diff
@@ -341,29 +388,212 @@ def start_buffer(self):
+        """
+        Method that converts the fixed-point buffer data to floating point data.
+        The provided data is converted as much as possible, but there are some
+        requirements to the data if all provided columns are to be converted; if
+        a key in the provided data cannot be converted it will be omitted in the
+        returned data or an exception will be raised, depending on the value of
+        raise_error.
+
+        The requirements for converting the data are as follows:
+        - Converting X, Y, magnitude and noise requires sensitivity data, which
+            can either be part of the provided data or can be provided via the
+            sensitivity argument
+        - The same holds for X2, Y2, magnitude2 and noise2 with sensitivity2.
+        - Converting the frequency requires both 'frequency part 1' and
+            'frequency part 2'.
+
+        :param dict buffer_data:
+            The data to be converted. Must be in the format as returned by the
+            `get_buffer` method: a dict of numpy arrays.
+
+        :param sensitivity:
+            If provided, the sensitivity used to convert X, Y, magnitude and noise.
+            Can be provided as a float or as an array that matches the length elements
+            in the buffer_data. If both a sensitivity is provided and present in the
+            buffer_data, the provided value is used for the conversion, but the
+            sensitivity in the buffer_data is stored in the returned dict.
+
+        :param sensitivity2:
+            Same as the first sensitivity argument, but for X2, Y2, magnitude2 and noise2.
+
+        :param bool raise_error:
+            Boolean that determines whether an exception is raised in case not all keys
+            provided in buffer_data can be converted. If False, the columns that cannot
+            be converted as omitted in the returned dict.
+
+        """
+        data = {}
+
+        # Sensitivity (for both single and dual modes)
+        for key in ["sensitivity", "sensitivity2"]:
+            if key in buffer_data:
+                data[key] = np.array([
+                    self.SENSITIVITIES[v % 32] * self.SEN_MULTIPLIER[v // 32]
+                    for v in buffer_data[key]
+                ])
+
+        # X, Y, magnitude, and noise data
+        if any(["x" in buffer_data,
+                "y" in buffer_data,
+                "magnitude" in buffer_data,
+                "noise" in buffer_data, ]):
+            # Requires a sensitivity
+            if sensitivity is not None:
+                pass
+            elif "sensitivity" in data:
+                sensitivity = data["sensitivity"]
+            elif raise_error:
+                raise ValueError("X, Y, magnitude and noise cannot be converted as no "
+                                 "sensitivity is provided, neither as argument or as "
+                                 "part of the buffer_data. ")
+
+            if sensitivity is not None:
```

Comment:
I think you can skip this check? Either sensitivity is already something (first if clause above), or it gets assigned from data, or we raise so never reach this code line.

Reply:
No not really, in case there is no sensitivity provided but `raise_error=False` is also passed, sensitivity will be None in the second if clause; you'd want to use some sort of if-else-finally logic, but that doesn't exist in python.

Reply:
Ah, sure, d'oh!